Add plugin: LaTeX Auto Complete

aiksxd opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/aiksxd/latex-autocomplete

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [ ] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
[containerEl.createEl('h2', { text: t.settings_title });](https://github.com/aiksxd/latex-autocomplete/blob/f4359641dca2e3b3c12eb5e6e751e7c29902e062/main.js#L692)
[Don't add a top-level heading in the settings tab, such as "General", "Settings", or the name of your plugin.](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)

["latex_mappings": "LaTeX Mappings",](https://github.com/aiksxd/latex-autocomplete/blob/f4359641dca2e3b3c12eb5e6e751e7c29902e062/main.js#L603), ["restore_defaults": "Restore Defaults",](https://github.com/aiksxd/latex-autocomplete/blob/f4359641dca2e3b3c12eb5e6e751e7c29902e062/main.js#L605), ["usage_title": "Usage Guide",](https://github.com/aiksxd/latex-autocomplete/blob/f4359641dca2e3b3c12eb5e6e751e7c29902e062/main.js#L608), ["example4_output": "Cancel completion",](https://github.com/aiksxd/latex-autocomplete/blob/f4359641dca2e3b3c12eb5e6e751e7c29902e062/main.js#L622), ["preview_mappings": "Preview Symbols Mappings",](https://github.com/aiksxd/latex-autocomplete/blob/f4359641dca2e3b3c12eb5e6e751e7c29902e062/main.js#L625)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI), product / brand names follow the casing set by the respective owner.

[textArea.inputEl.style.width = '100%';](https://github.com/aiksxd/latex-autocomplete/blob/f4359641dca2e3b3c12eb5e6e751e7c29902e062/main.js#L731-L733), [this.errorDisplay.style.color = 'var(--text-error)';](https://github.com/aiksxd/latex-autocomplete/blob/f4359641dca2e3b3c12eb5e6e751e7c29902e062/main.js#L740-L741), [previewTextArea.inputEl.style.width = '100%';](https://github.com/aiksxd/latex-autocomplete/blob/f4359641dca2e3b3c12eb5e6e751e7c29902e062/main.js#L782-L784), [this.previewErrorDisplay.style.color = 'var(--text-error)';](https://github.com/aiksxd/latex-autocomplete/blob/f4359641dca2e3b3c12eb5e6e751e7c29902e062/main.js#L791-L792), [addStyles() {](https://github.com/aiksxd/latex-autocomplete/blob/f4359641dca2e3b3c12eb5e6e751e7c29902e062/main.js#L873-L911)
You should avoid assigning styles via JavaScript or in HTML and instead move all these styles into CSS so that they are more easily adaptable by themes and snippets.

[usageContainer.createEl('h3', { text: t.usage_title });](https://github.com/aiksxd/latex-autocomplete/blob/f4359641dca2e3b3c12eb5e6e751e7c29902e062/main.js#L815), [examplesContainer.createEl('h3', { text: t.examples_title });](https://github.com/aiksxd/latex-autocomplete/blob/f4359641dca2e3b3c12eb5e6e751e7c29902e062/main.js#L826)
For section headings in settings use:
```ts
new Setting(containerEl).setName('name here').setHeading();
```

aiksxd commented:
> [containerEl.createEl('h2', { text: t.settings_title });](https://github.com/aiksxd/latex-autocomplete/blob/f4359641dca2e3b3c12eb5e6e751e7c29902e062/main.js#L692) [Don't add a top-level heading in the settings tab, such as "General", "Settings", or the name of your plugin.](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)
> 
> ["latex_mappings": "LaTeX Mappings",](https://github.com/aiksxd/latex-autocomplete/blob/f4359641dca2e3b3c12eb5e6e751e7c29902e062/main.js#L603), ["restore_defaults": "Restore Defaults",](https://github.com/aiksxd/latex-autocomplete/blob/f4359641dca2e3b3c12eb5e6e751e7c29902e062/main.js#L605), ["usage_title": "Usage Guide",](https://github.com/aiksxd/latex-autocomplete/blob/f4359641dca2e3b3c12eb5e6e751e7c29902e062/main.js#L608), ["example4_output": "Cancel completion",](https://github.com/aiksxd/latex-autocomplete/blob/f4359641dca2e3b3c12eb5e6e751e7c29902e062/main.js#L622), ["preview_mappings": "Preview Symbols Mappings",](https://github.com/aiksxd/latex-autocomplete/blob/f4359641dca2e3b3c12eb5e6e751e7c29902e062/main.js#L625) [Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI), product / brand names follow the casing set by the respective owner.
> 
> [textArea.inputEl.style.width = '100%';](https://github.com/aiksxd/latex-autocomplete/blob/f4359641dca2e3b3c12eb5e6e751e7c29902e062/main.js#L731-L733), [this.errorDisplay.style.color = 'var(--text-error)';](https://github.com/aiksxd/latex-autocomplete/blob/f4359641dca2e3b3c12eb5e6e751e7c29902e062/main.js#L740-L741), [previewTextArea.inputEl.style.width = '100%';](https://github.com/aiksxd/latex-autocomplete/blob/f4359641dca2e3b3c12eb5e6e751e7c29902e062/main.js#L782-L784), [this.previewErrorDisplay.style.color = 'var(--text-error)';](https://github.com/aiksxd/latex-autocomplete/blob/f4359641dca2e3b3c12eb5e6e751e7c29902e062/main.js#L791-L792), [addStyles() {](https://github.com/aiksxd/latex-autocomplete/blob/f4359641dca2e3b3c12eb5e6e751e7c29902e062/main.js#L873-L911) You should avoid assigning styles via JavaScript or in HTML and instead move all these styles into CSS so that they are more easily adaptable by themes and snippets. (`styles.css` file)
> 
> [usageContainer.createEl('h3', { text: t.usage_title });](https://github.com/aiksxd/latex-autocomplete/blob/f4359641dca2e3b3c12eb5e6e751e7c29902e062/main.js#L815), [examplesContainer.createEl('h3', { text: t.examples_title });](https://github.com/aiksxd/latex-autocomplete/blob/f4359641dca2e3b3c12eb5e6e751e7c29902e062/main.js#L826) For section headings in settings use:
> 
> ```ts
> new Setting(containerEl).setName('name here').setHeading();
> ```
Thank you for your review and the helpful feedback on my plugin submission.

I've addressed all the requested changes. Here's a summary of what was updated:

* Removed top-level headings in the settings tab
* Adjusted UI text to follow proper sentence casing
* Moved all styles from JavaScript to `styles.css`
* Used the recommended `new Setting(...).setHeading()` for section titles

You can find the full changelog here:
**Full Changelog**: https://github.com/aiksxd/latex-autocomplete/compare/1.0.0...1.1.0

Could you please review the updated submission at your convenience? Thank you again for your time and guidance :)
